mne/preprocessing/eyetracking/eyetracking.py

```diff
@@ -164,3 +170,137 @@ def _convert_mm_to_m(array):
+
+
+def convert_units(inst, calibration, to="radians"):
+    """Convert Eyegaze data from pixels to radians of visual angle or vice versa.
+
+    Parameters
+    ----------
+    inst : instance of Raw, Epochs, or Evoked
+        The Raw, Epochs, or Evoked instance with eyegaze channels.
+    calibration : Calibration
+        Instance of  Calibration, containing information about the screen size
+        (in meters), viewing distance (in meters), and the screen resolution
+        (in pixels).
+    to : str
+        Must be either ``"radians"`` or ``"pixels"``, indicating the desired unit.
+
+    Returns
+    -------
+    inst : instance of Raw | Epochs | Evoked
+        The Raw, Epochs, or Evoked instance, modified in place.
+
+    Notes
+    -----
+    .. versionadded:: 1.7
+    """
+    _validate_type(inst, (BaseRaw, BaseEpochs, Evoked), "inst")
+    _validate_type(calibration, Calibration, "calibration")
+    _check_option("to", to, ("radians", "pixels"))
+    _check_calibration(calibration)
+
+    # get screen parameters
+    screen_size = calibration["screen_size"]
+    screen_resolution = calibration["screen_resolution"]
+    dist = calibration["screen_distance"]
+
+    # loop through channels and convert units
+    converted_chs = []
+    for ch_dict in inst.info["chs"]:
+        if ch_dict["coil_type"] == FIFF.FIFFV_COIL_EYETRACK_POS:
+            unit = ch_dict["unit"]
+            name = ch_dict["ch_name"]
+
+            if ch_dict["loc"][4] == -1:  # x-coordinate
+                size = screen_size[0]
+                res = screen_resolution[0]
+            elif ch_dict["loc"][4] == 1:  # y-coordinate
+                size = screen_size[1]
+                res = screen_resolution[1]
+            else:
+                raise ValueError(
+                    f"loc array not set properly for channel '{name}'. Index 4 should"
+                    f"  be -1 or 1, but got {ch_dict['loc'][4]}"
+                )
+            # check unit, convert, and set new unit
+            if to == "radians":
+                if unit != FIFF.FIFF_UNIT_PX:
+                    raise ValueError(
+                        f"Data must be in pixels in order to convert to radians."
+                        f" Got {unit} for {name}"
+                    )
+                inst.apply_function(
+                    _pix_to_rad_1d, picks=name, size=size, res=res, dist=dist
+                )
+                ch_dict["unit"] = FIFF.FIFF_UNIT_RAD
+            elif to == "pixels":
+                if unit != FIFF.FIFF_UNIT_RAD:
+                    raise ValueError(
+                        f"Data must be in radians in order to convert to pixels."
+                        f" Got {unit} for {name}"
+                    )
+                inst.apply_function(
+                    _rad_to_pix_1d, picks=name, size=size, res=res, dist=dist
+                )
+                ch_dict["unit"] = FIFF.FIFF_UNIT_PX
+            converted_chs.append(name)
```

Comment:
+1 for warning if any of the gazes are > +/- 30 deg. Or at least trying it on some datasets to see if it's okay